app/models/cohort_columns/base.rb

```diff
@@ -22,6 +22,15 @@ class Base < ::ModelForm
     attribute :cohort_client
     attribute :editable, Boolean, lazy: false, default: true
     attribute :current_user
+    attribute :cohort_column, GrdaWarehouse::Cohorts::CohortColumn, lazy: true, default: ->(r, _) { GrdaWarehouse::Cohorts::CohortColumn.find_by!(class_name: r.class_name) }
```

Comment:
Do we need to pass `r` here?  Most of the time we should be interacting with the child class right?  So maybe?
```suggestion
    attribute :cohort_column, GrdaWarehouse::Cohorts::CohortColumn, lazy: true, default: -> { GrdaWarehouse::Cohorts::CohortColumn.find_by!(class_name: class_name) }
```

Or maybe even:
```suggestion
    def cohort_column
      GrdaWarehouse::Cohorts::CohortColumn.find_by!(class_name: r.class_name)
    end
```

I'm probably missing something, but that should mean we don't actually need to set it up in the factories.

Reply:
I can give the latter suggesting a go and run through some tests. I ran into a handful of issues getting this to run tests properly, so if this possibly clears up some hurdles, it makes sense to me